plugins/node/opentelemetry-instrumentation-memcached/src/instrumentation.ts

```diff
@@ -0,0 +1,186 @@
+          );
+          if (moduleExports === undefined) return;
+          // `command` is documented API missing from the types
+          this._unwrap(moduleExports.prototype, 'command' as keyof Memcached);
+        }
+      ),
+    ];
+  }
+
+  wrapCommand(
+    moduleVersion: undefined | string,
+    original: (
+      queryCompiler: () => Memcached.CommandData,
+      server?: string
+    ) => any
+  ) {
+    const instrumentation = this;
+    return function (
+      this: Memcached,
+      queryCompiler: () => Memcached.CommandData,
+      server?: string
+    ) {
+      if (typeof queryCompiler !== 'function') {
+        return original.apply(this, arguments as any);
+      }
+      // The name will be overwritten later
+      const span = instrumentation.tracer.startSpan(
+        'unknown memcached command',
+        {
+          kind: api.SpanKind.CLIENT,
+          attributes: {
+            'memcached.version': moduleVersion,
+            ...Instrumentation.COMMON_ATTRIBUTES,
+          },
+        }
+      );
+      const context = api.trace.setSpan(api.context.active(), span);
+
+      return api.context.with(
+        context,
+        original,
+        this,
+        instrumentation.wrapQueryCompiler.call(
+          instrumentation,
+          queryCompiler,
+          this,
+          server,
+          context,
+          span
+        ),
+        server
+      );
+    };
+  }
+
+  wrapQueryCompiler(
+    original: () => Memcached.CommandData,
+    client: Memcached,
+    server: undefined | string,
+    context: api.Context,
+    span: api.Span
+  ) {
+    const instrumentation = this;
+    return function (this: Memcached) {
+      const query = original.apply(this, arguments as any);
+      const callback = query.callback;
+
+      span.updateName(`${query.type} ${query.key}`);
+      span.setAttributes({
+        'db.memcached.key': query.key,
+        'db.memcached.lifetime': query.lifetime,
+        [SemanticAttributes.DB_OPERATION]: query.type,
+        'db.statement': (instrumentation._config as InstrumentationConfig)
+          .collectCommand
+          ? query.command
+          : undefined,
+        ...utils.getPeerAttributes(client, server, query),
+      });
+
+      query.callback = api.context.bind(
+        context,
+        function (this: Memcached.CommandData, err: any) {
+          if (err) {
+            span.recordException(err);
+            span.setStatus({
+              code: api.SpanStatusCode.ERROR,
+              message: err.message,
+            });
+          }
+
+          span.end();
+
+          if (typeof callback === 'function') {
+            return callback.apply(this, arguments as any);
+          }
+        }
+      );
+
+      return query;
+    };
+  }
+
+  private ensureWrapped(
+    moduleVersion: string | undefined,
+    obj: any,
+    methodName: string,
+    wrapper: (original: any) => any
+  ) {
+    api.diag.debug(
```

Comment:
you should be able to use `this._diag` so the logger is namespaced with the instrumentation name

Reply:
Fantastic! Thanks!